src/CameraParameters.cpp

```diff
@@ -97,6 +97,43 @@ bool CameraParameters::setParameter(std::string key, std::string value)
     return true;
 }
 
+bool CameraParameters::setParameter(std::string param_id, float param_value)
+{
+    char str[CAM_PARAM_VALUE_LEN];
+    cam_param_union u;
+    u.param_float = param_value;
+    memcpy(&str[0], &u.param_float, sizeof(float));
+    std::string str2(str, sizeof(str));
+    return setParameter(param_id, str2);
+}
+bool CameraParameters::setParameter(std::string param_id, uint32_t param_value)
+{
+    char str[CAM_PARAM_VALUE_LEN];
+    cam_param_union u;
+    u.param_uint32 = param_value;
+    memcpy(&str[0], &u.param_uint32, sizeof(uint32_t));
+    std::string str2(str, sizeof(str));
```

Comment:
This usage of c strings and c++ strings is incorrect. And you are going to lose information in some cases. And it is really hard to understand what are the reasons to do this.

For example if param_value is 0x4800 you are going to copy all those bytes to str (0x00, 0x48, 0x00, 0x00 - little endian) and than copy those bytes to str2 data. In CameraComponent_V4L2::getParam function you are going to call c_str() in str2.
As str2 first byte is 0, the c_str will be an empty string and no bytes will be copied.

Reply:
Why? The idea is to convert the parameter value from a data type to byte array and then store it as string.
It works, also I have a sample test app to test set/get operations. I will push that as one of the samples. Nevertheless, I have made changes to cut corner cases and simplify. Kindly review.

Reply:
I haven't seen the new code you updated, but I still believe that when using the code I reviewed you are going to lose information for some values of param_value and get invalid access memory for other values. Have you tried to use param_value=0x4800 as I suggested? As soon as you have a sample app that make it easy to test your data conversions, let me know.

Reply:
I tried 0x4800 and it works. I have pushed the sample app(basic), you can try it out.